src/fides/api/ops/service/connectors/saas/authenticated_client.py

```diff
@@ -37,18 +43,14 @@ def __init__(
         self,
         uri: str,
         configuration: ConnectionConfig,
-        request_client_config: Optional[ClientConfig] = None,
+        saas_request: Optional[SaaSRequest] = None,
     ):
-        saas_config = configuration.get_saas_config()
+        self.saas_config = configuration.get_saas_config()
         self.configuration = configuration
+        self.saas_request = saas_request
         self.session = Session()
         self.uri = uri
         self.key = configuration.key
-        self.client_config = (
-            request_client_config
-            if request_client_config
-            else saas_config.client_config  # type: ignore
-        )
```

Comment:
I found the logic here a bit confusing before. My thought with this change is that instead of passing in a `ClientConfig` we just have access to the `SaaSRequest`. Currently we pull `client_config` and `ignore_errors` from `saas_request` and with this change also need `rate_limits`.